### PR TITLE
Change default fractional_seaice option from 0 to 1

### PR DESCRIPTION
**Pull request by weiwangncar**
TYPE: enhancement

KEYWORDS: namelist fractional_seaice, default, replacing original PR#1128

SOURCE: Internal

DESCRIPTION OF CHANGES:
Since GFS, the most commonly used input data source, has been providing fractional seaice
data for a while, it seems reasonable to make the default option for fractional_seaice to be 1.
This should also help our general users. Both ERA5 and ERA-Interim provide fractional seaice
data, too. 

Turning on the fractional seaice flag broke many MPI vs OpenMP comparison tests. This led to the finding that the 
option is not thread safe for a number surface layer physics options: revised MM5, original MM5, MYJ, QNSE and PX (sf_sfclay_physics = 1, 91, 2, 4, and 7). This PR additionally provides the fix for these options.

LIST OF MODIFIED FILES:
M Registry/Registry.EM_COMMON
M phys/module_surface_driver.F

TESTS CONDUCTED:
1. Regression test all pass.

RELEASE NOTE: The default option for using fractional_seaice has been changed from 0 (off) to 1 (on) starting with release v4.3. In addition, several surface layer schemes are made thread safe if fractional seaice option is used. These are revised MM5, original MM5, MYJ, QNSE and PX surface layer scheme (sf_sfclay_physics = 1, 91, 2, 4, and 7).

### CODE REVIEW

**Comment by davegill** [comment]
@weiwangncar 
Wei,
Just artificially bumping up your PR numbers, I see. :)

**Comment by weiwangncar** [comment]
@davegill I wanted to hit #1400!

**Comment by davegill** [comment]
@weiwangncar 
Good detective work!
Once the jenkins test is back, I'll approve.

File: phys/module_surface_driver.F [hunk 1]
```diff
@@ -1994,7 +1992,7 @@ SUBROUTINE surface_driver(                                         &
                  ids,ide, jds,jde, kds,kde,                          &
                  ims,ime, jms,jme, kms,kme,                          &
                  i_start(ij),i_end(ij), j_start(ij),j_end(ij), kts,kte,    &
-                 ustm,ck,cka,cd,cda,isftcflx,iz0tlnd,                &
+                 ustm,ck,cka,cd,cda,isftcflx,iz0tlnd,scm_force_flux, &
```
Comment (davegill) [inline phys/module_surface_driver.F:1995]: @weiwangncar 
I see this `scm_force_flux` has been added all over the place in subroutine calls. What is it for in this PR?

Comment (weiwangncar) [reply phys/module_surface_driver.F:1995]: @davegill This is in the call to SFCLAY, and SFCLAYREV, so I thought I'll pass it through.

**Review by davegill — APPROVED** [review]

**Comment by davegill** [comment]
:shipit:

**Pull request merged**